ydb/docs/en/core/dev/example-app/_includes/example-cpp.md

```diff
@@ -391,44 +409,47 @@ Episode 3, Season: 5, title: Chief Operating Officer, Air date: Sun Apr 08, 2018
 Code snippet for `BeginTransaction` and `tx.Commit()` calls:
 
 ```c++
-// Show usage of explicit Begin/Commit transaction control calls.
-// In most cases it's better to use transaction control settings in ExecuteDataQuery calls instead
-// to avoid additional hops to YDB cluster and allow more efficient execution of queries.
-static TStatus ExplicitTclTransaction(TSession session, const TString& path, const TInstant& airDate) {
-    auto beginResult = session.BeginTransaction(TTxSettings::SerializableRW()).GetValueSync();
-    if (!beginResult.IsSuccess()) {
-        return beginResult;
-    }
-
-    // Get newly created transaction id
-    auto tx = beginResult.GetTransaction();
+void ExplicitTcl(TQueryClient client, const TString& path) {
+    // Show usage of explicit Begin/Commit transaction control calls.
+    // In most cases it's better to use transaction control settings in ExecuteDataQuery calls instead
+    // to avoid additional hops to YDB cluster and allow more efficient execution of queries.
+    ThrowOnError(client.RetryQuerySync([&path](TQueryClient client) -> TStatus {
+        auto airDate = TInstant::Now();
+        auto session = client.GetSession().GetValueSync().GetSession();
```

Comment:
Why `GetSession()` is called twice? The first one probably returns a promise; hence, `GetValueSync()` is needed, but what does the second `GetSession()` do?

Reply:
GetValueSync() returns TCreateSessionResult that contains session which can be obtained from GetSession method